docs/topics/type-safe-builders.md

```diff
@@ -242,7 +242,46 @@ html {
 }
 ```
 
-## Full definition of the com.example.html package
+You can also apply the `@DslMarker` annotation directly to [function types](lambdas.md#function-types).
+To do so, annotate a `@DslMarker` annotation with `@Target(AnnotationTarget.TYPE)`:
+
+```kotlin
+@Target(AnnotationTarget.TYPE)
+@DslMarker
+annotation class HtmlTagMarker
+```
+
+This allows the `@DslMarker` annotation to be applied to function types, most commonly to lambdas with receivers. For example:
+
+```kotlin
+fun HTML.head(init: @HtmlTagMarker Head.() -> Unit): Head {
+    val head = Head()
+    head.init()
+    return head
+}
+
+fun Head.title(init: @HtmlTagMarker Title.() -> Unit): Title {
+    val title = Title()
+    title.init()
+    return title
+}
+```
+
+When you call these functions, the `@DslMarker` annotation restricts access to outer receivers unless you specify them explicitly:
```

Comment:
Maybe explain this a bit, as in "...annotation restricts access to outer receivers [within the body of a `@DslMarker`-marked lambdas] unless you...?